src/libraries/Microsoft.PowerFx.Core/Public/Values/RecordValue.cs

```diff
@@ -161,5 +161,10 @@ public sealed override void Visit(IValueVisitor visitor)
         {
             visitor.Visit(this);
         }
+
+        public virtual void UpdateFields(IEnumerable<KeyValuePair<string, FormulaValue>> record)
```

Comment:
>void [](http://example.com/codeflow?start=23&length=4)

Patch() can fail. So this may need return a FormulaValue (like Append did) 

Try this in PowerApps, it would be a good test case:

IfError(Patch(Table({f2:5}), {f2:5}, { f2:6}), Notify(FirstError.Message))

Reply:
This should probably be an Error Value (not exception), since it can happen at runtime. 

An exception is for SDK level errors and will abort the entire execution (and not return the error to the expression).